source/common/network/transport_socket_options_impl.h

```diff
@@ -27,12 +55,16 @@ class TransportSocketOptionsImpl : public TransportSocketOptions {
   const std::vector<std::string>& applicationProtocolListOverride() const override {
     return override_alpn_list_;
   }
+  const std::vector<std::string>& applicationProtocolListFallback() const override {
+    return alpn_fallback_list_;
+  }
   void hashKey(std::vector<uint8_t>& key) const override;
 
 private:
   const absl::optional<std::string> override_server_name_;
   const std::vector<std::string> override_verify_san_list_;
   const std::vector<std::string> override_alpn_list_;
+  const std::vector<std::string> alpn_fallback_list_;
```

Comment:
I think this is related to your reviewer comment, but tbh I'm pretty confused about override vs. fallback. Can you add more comments or maybe we should discuss your reviewer comment before proceeding?